contracts/core/VaultUtils.sol

```diff
@@ -204,76 +202,39 @@ contract VaultUtils is IVaultUtils, Constants {
                 remainingCollateral = position.collateral + delta;
             }
 
-            return _checkMaxThreshold(remainingCollateral, position.size, migrateFeeUsd, _indexToken, _raise);
+            return _checkMaxThreshold(remainingCollateral, position.size, migrateFeeUsd, position.indexToken, _raise);
         } else {
             return (LIQUIDATE_NONE_EXCEED, 0);
         }
     }
 
+    function validateMinLeverage(uint256 _size, uint256 _collateral) external pure override {
+        require(_size >= _collateral, "leverage cannot be less than 1");
+    }
+
     function validateMaxLeverage(address _indexToken, uint256 _size, uint256 _collateral) external view override {
         require(
-            _collateral * (priceManager.maxLeverage(_indexToken) + LEVERAGE_SLIPPAGE) >= _size * MIN_LEVERAGE,
-            "Vault: maxLeverage exceeded"
+            _size * 10 ** 4 <= _collateral * (priceManager.maxLeverage(_indexToken) + LEVERAGE_SLIPPAGE),
+            "maxLeverage exceeded"
         );
     }
 
-    function validatePosData(
```

Comment:
need to check this removed logic is fully implemented

Reply:
I have removed it as it's fully implemented in a simpler way